types/http-assert/index.d.ts

```diff
@@ -11,30 +11,30 @@
  * @param msg the message of the error, defaulting to node's text for that status code
  * @param opts custom properties to attach to the error object
  */
-declare function assert(value: any, status?: number, msg?: string, opts?: {}): void;
-declare function assert(value: any, status?: number, opts?: {}): void;
+declare function assert(value: any, status?: number, msg?: string, opts?: Record<string, any>): asserts value;
+declare function assert(value: any, status?: number, opts?: Record<string, any>): asserts value;
 
 declare namespace assert {
     /**
      * @param status the status code
      * @param msg the message of the error, defaulting to node's text for that status code
      * @param opts custom properties to attach to the error object
      */
-    type Assert = <T>(a: T, b: T, status?: number, msg?: string, opts?: {}) => void;
+    type Assert = <T>(a: T, b: T, status?: number, msg?: string, opts?: Record<string, any>) => void;
```

Comment:
Shouldn't return type here and few lines below not be changed to `asserts a`?

Reply:
This tests shallow equality between `a` and `b`. `asserts a` means it would test if `a` is truthy.

This type may be enhanced somehow, but I’m not sure how at this point.

Reply:
You are right. This is not solvable at the moment or at least I don't know how...